temp/legacy-code/src/main/java/com/ivy/legacy/datamodel/temp/TransactionExt.kt

```diff
@@ -14,6 +15,14 @@ fun Transaction.toLegacy(mapper: TransactionMapper): LegacyTransaction {
     return with(mapper) { toEntity().toLegacyDomain() }
 }
 
+suspend fun LegacyTransaction.toDomain(mapper: TransactionMapper): Transaction? {
+   return with(mapper) {
+       toEntity().toDomain().getOrNull()?.let {
+           it
```

Comment:
The `?.let { it }` part seems useless, let's remove it

Reply:
Okay, can you help me here then, untill unless i use its context, i can't get Legacy Transaction return type,
and i observed in other places, that for transforming from Legacy Version to New Version for ex Legacy Transaction to Transaction, we are directly converting it in the codes, 
So i created an extension function here, but i haven't used it anywhere, so should i call it then.

Reply:
That's weird, it should work without the let cuz it `let` literally does nothing

Reply:
well, will check it again, and any suggestions whether we can use this extension or not, 
because as of now we don't have any extensions to convert from Legacy Data to New Data one.

Reply:
Let's use it and when we get rid of the legacy model we can delete it

Reply:
Sure got it. Will make the changes and do the final commit.